ass3.cpp

```diff
@@ -0,0 +1,95 @@
+#include <iostream>
+#include <cstring>
+#include <set>
+#include <vector>
+
+using namespace std;
+
+#define ROW_SIZE 2
+#define COL_SIZE 3
+
+
+class Dictionary {
+    set<string> dict_words;
+    set<string> dict_prefixes;
+public:
+    Dictionary() {};
+    
+    void fill_dict(vector<string> words) {
+        for (int i = 0; i < words.size(); ++i) {
+            dict_words.insert(words[i]);
+            
+            string prefix = "";
+            for (int j = 0; j < words[i].length(); ++j) {
+                prefix += words[i][j];
+                dict_prefixes.insert(prefix);
+            }
+        }
+    }
+
+
+    bool isWord(string s) {
+        return dict_words.find(s) != dict_words.end();
+    }
+    
+    bool isPrefix(string s) {
+        return dict_prefixes.find(s) != dict_prefixes.end();
+    }
+};
+
+void find_pref_word(char table[ROW_SIZE][COL_SIZE], vector<vector<bool> > visited, int i, int j, string &str, Dictionary dict, set<string> &result_words) {
```

Comment:
I think the table should be `vector<vector<char>>` or `vector<string>`.
If you want, you can use `char *` or `char **` and pass the sizes as parameters.